src/main.rs

```diff
@@ -194,10 +200,13 @@ where
     active_threads.push(
         thread::Builder::new()
             .name("sniffing_handler".to_string())
-            .spawn(move || {
-                while running.load(Ordering::Acquire) {
+            .spawn(move || 'sniffing: loop {
+                for sniffer in sniffers.iter_mut() {
                     if let Some(segment) = sniffer.next() {
-                        network_utilization.lock().unwrap().update(&segment)
+                        network_utilization.lock().unwrap().update(segment);
+                    }
+                    if !running.load(Ordering::Acquire) {
```

Comment:
This is a little problematic. Now we would be acquiring the AtomicBool just for the duration of this scope (the current loop iteration), meaning that we can potentially listen for another network frame after the program has stopped running. Why do you think this needs to be changed?

Reply:
Correct me if I'm wrong, but I think that `load` [does not return a guard](https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html#method.load) here and just atomically reads the value. The `Acquire` ordering would be an indication for the compiler as to how access to the boolean from competing threads should be ordered. 

The intent behind my change was to ensure that we check if the program is running between peeks to each network interface (similar as to what we were doing before) as opposed to iterating through all the interfaces before checking the flag.

Reply:
Ahh, that's right - it doesn't return a guard. I was confusing it with a `Mutex`.